### PR TITLE
feat: upload find ad form

### LINKED ISSUE #57: Upload find ad form
Form for uploading a find ad.

### PR DESCRIPTION
**Pull request by Kal-Elx**
Closes #57

### CODE REVIEW

File: lib/widgets/upload/ad_form_title.dart [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+class AdFormTitle extends StatelessWidget {
+  final String title;
+
+  AdFormTitle(this.title);
+
+  @override
+  Widget build(BuildContext context) {
+    return Text(title);
+  }
+}
```
Comment (johanehinger) [inline lib/widgets/upload/ad_form_title.dart:15]: Is this really necessary? All it does is return Text widget might as well use the Text widget directly.
https://www.youtube.com/watch?v=-AQfQFcXac8&ab_channel=javidx9

File: lib/widgets/upload/ad_form_text_field.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    this.hint,
+    this.optional = false,
+    this.lines = 1,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final l10n = MyLocalizations.of(context);
+
+    return TextFormField(
```
Comment (johanehinger) [inline lib/widgets/upload/ad_form_text_field.dart:20]: This one makes sense to extract, good!

**Review by johanehinger — CHANGES_REQUESTED** [review]
The approach is correct! But don't get caught in overdoing things. Flutter already contains good widgets and we should strive to use them when possible in order for other developers to understand our code more easaly.

File: lib/widgets/upload/ad_form_title.dart (hunk 1 above)
Comment (Kal-Elx) [reply lib/widgets/upload/ad_form_title.dart:15]: My idea is that we can extend this widget later with design, padding etc.

**Review by johanehinger — APPROVED** [review]
Looks good

**Pull request merged**